src/projection/rh_ydown.rs

```diff
@@ -358,7 +358,99 @@ pub fn perspective_reversed_infinite_z_vk(
     Mat4::new(
         Vec4::new(sx, 0.0, 0.0, 0.0),
         Vec4::new(0.0, sy, 0.0, 0.0),
-        Vec4::new(0.0, 0.0, 0.0, -1.0),
+        Vec4::new(0.0, 0.0, 0.0, 1.0),
         Vec4::new(0.0, 0.0, z_near, 0.0),
     )
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::f32::consts::FRAC_PI_2;
+
+    fn vk_corners() -> [(f32, f32, f32); 4] {
+        [
+            (-1.0, -1.0, 0.0),
+            (1.0, 1.0, 0.0),
+            (-1.0, 1.0, 1.0),
+            (1.0, -1.0, 1.0),
+        ]
+    }
+
+    fn eq_viewport_corners(
+        perspective: Mat4,
+        corners: [(f32, f32, f32); 4],
+        expected: [(f32, f32, f32); 4],
+    ) -> bool {
+        for (corner, &expected) in corners.iter().zip(expected.iter()) {
+            let actual = Vec3::from_homogeneous_point(
+                perspective * Vec3::from(corner).into_homogeneous_point(),
+            );
+            let expected = Vec3::from(expected);
+            if (actual - expected).mag() > 0.01 {
```

Comment:
you should be able to use `util::equals_eps` here :)

Reply:
I initially tried that, but the diagnostics looked like "`0.75 should equal 1 with epsilon 0.01 but doesn't.`" instead of "`Vec3 { x: 0.75, y: 0.75, z: 0.35714293 } should equal Vec3 { x: 1.0, y: 1.0, z: 0.0 } with epsilon 0.01 but doesn't.`" which made it too difficult to troubleshoot the matrices. I thought about trying to change the `EqualsEps` implementations for vectors to show the entire vector, but ran into difficulties figuring out an implementation that worked both for regular vectors and for wide vectors.